core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java

```diff
@@ -384,11 +388,157 @@ public static TableEntity fromTablePO(TablePO tablePO, Namespace namespace) {
+          .withCurrentVersion(INIT_VERSION)
+          .withLastVersion(INIT_VERSION)
+          .withDeletedAt(DEFAULT_DELETED_AT)
+          .withFilesetVersionPO(filesetVersionPO)
+          .build();
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Failed to serialize json object:", e);
+    }
+  }
+
+  /**
+   * Update FilesetPO version
+   *
+   * @param oldFilesetPO the old FilesetPO object
+   * @param newFileset the new FilesetEntity object
+   * @return FilesetPO object with updated version
+   */
+  public static FilesetPO updateFilesetPOWithVersion(
+      FilesetPO oldFilesetPO, FilesetEntity newFileset, boolean needUpdateVersion) {
+    try {
+      Long lastVersion = oldFilesetPO.getLastVersion();
+      Long currentVersion;
+      FilesetVersionPO newFilesetVersionPO;
+      // Will set the version to the last version + 1
+      if (needUpdateVersion) {
+        lastVersion++;
+        currentVersion = lastVersion;
+        newFilesetVersionPO =
+            new FilesetVersionPO.Builder()
+                .withMetalakeId(oldFilesetPO.getMetalakeId())
+                .withCatalogId(oldFilesetPO.getCatalogId())
+                .withSchemaId(oldFilesetPO.getSchemaId())
+                .withFilesetId(newFileset.id())
+                .withVersion(currentVersion)
+                .withFilesetComment(newFileset.comment())
+                .withStorageLocation(newFileset.storageLocation())
+                .withProperties(
+                    JsonUtils.anyFieldMapper().writeValueAsString(newFileset.properties()))
+                .withDeletedAt(DEFAULT_DELETED_AT)
+                .build();
+      } else {
+        currentVersion = oldFilesetPO.getCurrentVersion();
+        newFilesetVersionPO = oldFilesetPO.getFilesetVersionPO();
+      }
+      return new FilesetPO.Builder()
+          .withFilesetId(newFileset.id())
+          .withFilesetName(newFileset.name())
+          .withMetalakeId(oldFilesetPO.getMetalakeId())
+          .withCatalogId(oldFilesetPO.getCatalogId())
+          .withSchemaId(oldFilesetPO.getSchemaId())
+          .withType(newFileset.filesetType().name())
+          .withAuditInfo(JsonUtils.anyFieldMapper().writeValueAsString(newFileset.auditInfo()))
+          .withCurrentVersion(currentVersion)
+          .withLastVersion(lastVersion)
+          .withDeletedAt(DEFAULT_DELETED_AT)
+          .withFilesetVersionPO(newFilesetVersionPO)
+          .build();
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Failed to serialize json object:", e);
+    }
+  }
+
+  public static boolean checkFilesetVersionNeedUpdate(
+      FilesetVersionPO oldFilesetVersionPO, FilesetEntity newFileset) {
+    if (!oldFilesetVersionPO.getFilesetComment().equals(newFileset.comment())
+        || !oldFilesetVersionPO.getStorageLocation().equals(newFileset.storageLocation())) {
+      return true;
```

Comment:
Only these changes will cause the version to be updated. @jerryshao, could you assist in confirming it? should we just omit the `auditInfo`?

Reply:
I think we probably don't need `auditInfo` in the version table for two reasons:  
1. This parameter is automatically generated by the server and cannot be modified by users.  
2. We should always keep the latest `auditInfo`. Even if the user performs a rollback, we should update `auditInfo` to the time and role of the rollback instead of using the historical `auditInfo`.

Reply:
I see.

Reply:
I think `auditInfo` is not so necessary to maintain as multiple versions. I'm OK with the current implementation.